bindings/rust/s2n-tls-tokio/src/lib.rs

```diff
@@ -363,15 +363,19 @@ where
     fn poll_shutdown(mut self: Pin<&mut Self>, ctx: &mut Context<'_>) -> Poll<io::Result<()>> {
         ready!(self.as_mut().poll_blinding(ctx))?;
 
-        // s2n_shutdown must not be called again if it errors
+        // s2n_shutdown_send must not be called again if it errors
         if self.shutdown_error.is_none() {
             let result = ready!(self.as_mut().with_io(ctx, |mut context| {
-                context.conn.as_mut().poll_shutdown().map(|r| r.map(|_| ()))
+                context
+                    .conn
+                    .as_mut()
+                    .poll_shutdown_send()
+                    .map(|r| r.map(|_| ()))
             }));
             if let Err(error) = result {
                 self.shutdown_error = Some(error);
-                // s2n_shutdown reading might have triggered blinding again
-                ready!(self.as_mut().poll_blinding(ctx))?;
+                // s2n_shutdown_send only writes, so will never trigger blinding again.
+                // So we do not need to poll_blinding again after this error.
```

Comment:
I went back and forth on whether to leave the call to poll_blinding in or not. We could leave it in just in case we ever introduce a possible blinded error in s2n_shutdown_send, but that should never happen since we only blind when reading. In the end I removed it because I can't test it anymore without doing something crazy and undefined like calling s2n_recv from inside my mocked tcp write call. That's also why I deleted the "shutdown_with_blinding_bad_close_record" test-- I can't reasonably trigger that case anymore.

I left a comment as a reminder of the "double-blinding" problem just in case. But again, we should never read here, since it's the Async**Write** trait :)